presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -36,6 +37,13 @@
     private final PlanNode left;
     private final PlanNode right;
     private final List<EquiJoinClause> criteria;
+    /**
+     * List of output symbols produced by join. Output symbols
```

Comment:
Isn't this responsibility of `ValidateDependenciesChecker`?
For instance `ProjectNode` doesn't check the condition in constructor.

I will add `checkState` as separate commit